.github/workflows/regression-report.yml

```diff
@@ -122,11 +126,172 @@ jobs:
           # shellcheck disable=SC2005
           echo "$(cat "$GITHUB_OUTPUT")"
 
-  make_report:
+
+  handle_config_bundle:
+    needs: parse_comment
+    if: needs.parse_comment.outputs.mode == 'use_config_bundle'
     runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.set_links.outputs.config_link }}
+      projects_link: ${{ steps.set_links.outputs.projects_link }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set config and projects links
+        id: set_links
+        run: |
+          BASE_URL="https://raw.githubusercontent.com/checkstyle/test-configs/main"
+          CONFIG_BUNDLE="${{ needs.parse_comment.outputs.config_bundle }}"
+          CONFIG_LINK="$BASE_URL/$CONFIG_BUNDLE/config.xml"
+          PROJECTS_LINK="$BASE_URL/$CONFIG_BUNDLE/list-of-projects.properties"
+          ./.ci/append-to-github-output.sh "config_link" "$CONFIG_LINK"
+          ./.ci/append-to-github-output.sh "projects_link" "$PROJECTS_LINK"
+
+  handle_generate_config:
     needs: parse_comment
-    if: needs.parse_comment.outputs.config_link != ''
-      || needs.parse_comment.outputs.projects_link != ''
+    if: needs.parse_comment.outputs.mode == 'generate_config'
+    runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.generate_config.outputs.config_link }}
+      projects_link: ${{ steps.generate_config.outputs.projects_link }}
+    steps:
+      - name: Checkout PR codebase
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Checkout Checkstyle repository
+        uses: actions/checkout@v4
+        with:
+          repository: checkstyle/checkstyle
+          path: .ci-temp/checkstyle
+
+      - name: Find input file
+        id: find_input_file
+        run: |
+
+          INPUT_FILE="${{ needs.parse_comment.outputs.input_file }}"
+          CHECKSTYLE_DIR=".ci-temp/checkstyle"
+          FILENAME=$(basename "$INPUT_FILE")
+
+          # Remove .java extension if present
+          FILENAME_NO_EXT="${FILENAME%.java}"
```

Comment:
not clear why we need this, Input file can be any file with any extension.
just search by content that user provided.

Reply:
done